src/polycap.h

```diff
@@ -0,0 +1,105 @@
+// Libraries to include
+#include "config.h"
+#include <stdio.h>
+#ifdef _WIN32
+  #define _CRT_RAND_S // for rand_s -> see https://msdn.microsoft.com/en-us/library/sxtz2fa8.aspx
+#endif
+#include <stdlib.h>
+#include <math.h>
+#include <string.h>
+#include <stdint.h>
+#include <omp.h> /* openmp header */
+#include <limits.h>
+#include <xraylib.h>
+#ifdef HAVE_EASYRNG
+  #include <easy_rng.h>
+  #include <easy_randist.h>
+  typedef easy_rng_type polycap_rng_type;
+  typedef easy_rng polycap_rng;
+  #define polycap_rng_alloc(T) easy_rng_alloc(T)
+  #define polycap_rng_free(rng) easy_rng_free(rng)
+  #define polycap_rng_set(rng, seed) easy_rng_set(rng, seed)
+  #define polycap_rng_uniform(rng) easy_rng_uniform(rng)
+  #define polycap_rng_mt19937 easy_rng_mt19937
+#else
+  #include <gsl/gsl_rng.h>
+  #include <gsl/gsl_randist.h>
+  typedef gsl_rng_type polycap_rng_type;
+  typedef gsl_rng polycap_rng;
+  #define polycap_rng_alloc(T) gsl_rng_alloc(T)
+  #define polycap_rng_free(rng) gsl_rng_free(rng)
+  #define polycap_rng_set(rng, seed) gsl_rng_set(rng, seed)
+  #define polycap_rng_uniform(rng) gsl_rng_uniform(rng)
+  #define polycap_rng_mt19937 gsl_rng_mt19937
+#endif
+#include <complex.h> //complex numbers required for Fresnel equation (reflect)
+#include <gsl/gsl_multifit.h>
+#include <stdbool.h>
+
+// Structure definitions
+struct inp_file
+  {
+  double sig_rough;
+  double sig_wave;
+  double corr_length;
+  double d_source;
+  double d_screen;
+  double src_x;
+  double src_y;
+  double src_sigx;
+  double src_sigy;
+  double src_shiftx;
+  double src_shifty;
+  int nelem;
+  int iz[NELEM];
+  double wi[NELEM];
+  double density;
+  double e_start;
+  double e_final;
+  double delta_e;
+  int ndet;
+  int shape;
+  char prf[80];
```

Comment:
In de regel moet je altijd vermijden te werken met fixed size arrays, en zeker als het voor filenames is. Beter te werken met malloc hier.